FYI: fix buglet in gdbpy_get_display_hint
http://sourceware.org/ml/gdb-patches/2010-07/msg00190.html
http://sourceware.org/ml/gdb-cvs/2010-07/msg00061.html

### src/gdb/ChangeLog	2010/07/09 02:39:57	1.11979
### src/gdb/ChangeLog	2010/07/09 20:29:56	1.11980
## -1,3 +1,8 @@
+2010-07-09  Tom Tromey  <tromey@redhat.com>
+
+	* python/py-prettyprint.c (gdbpy_get_display_hint): Don't use
+	'hint' if it is NULL.
+
 2010-07-09  Hui Zhu  <teawater@gmail.com>
 
 	* source.c (print_source_lines_base): Add check for noprint.
--- src/gdb/python/py-prettyprint.c	2010/06/11 15:36:09	1.12
+++ src/gdb/python/py-prettyprint.c	2010/07/09 20:29:56	1.13
@@ -229,10 +229,12 @@
     return NULL;
 
   hint = PyObject_CallMethodObjArgs (printer, gdbpy_display_hint_cst, NULL);
-  if (gdbpy_is_string (hint))
-    result = python_string_to_host_string (hint);
   if (hint)
-    Py_DECREF (hint);
+    {
+      if (gdbpy_is_string (hint))
+	result = python_string_to_host_string (hint);
+      Py_DECREF (hint);
+    }
   else
     gdbpy_print_stack ();
 
